log/configmanager.go

```diff
@@ -1,238 +0,0 @@
-		case []byte:
-			json.Unmarshal(val, &loglevel)
-		}
-	}
-
-	return loglevel
-}
-
-//Initialize the component config
-func InitComponentConfig(componentLabel string, configType ConfigType, cm *ConfigManager) (*ComponentConfig, error) {
-	//construct ComponentConfig
-	//verify componentConfig for provided componentLabel and configType is present in etcd,
-
-	cConfig := &ComponentConfig{
-		componentLabel:   componentLabel,
-		configType:       configType,
-		CManager:         cm,
-		changeEventChan:  nil,
-		kvStoreEventChan: nil,
-		logLevel:         nil,
-	}
-
-	return cConfig, nil
-}
-
-type ComponentConfig struct {
-	CManager         *ConfigManager
-	componentLabel   string
-	configType       ConfigType
-	changeEventChan  chan *ConfigChangeEvent
-	kvStoreEventChan chan *kvstore.Event
-	logLevel         []LogLevel
-}
-
-type LogLevel struct {
-	PackageName string
-	Level       string
-}
-
-//MonitorForConfigChange watch on the keys
-//If any changes happen then process the event create new ConfigChangeEvent and return
-func (c *ComponentConfig) MonitorForConfigChange() (chan *ConfigChangeEvent, error) {
-	//call makeConfigPath function to create path
-	key := c.makeConfigPath()
-
-	//call backend createwatch method
-	c.kvStoreEventChan = make(chan *kvstore.Event)
-	c.changeEventChan = make(chan *ConfigChangeEvent)
-
-	c.kvStoreEventChan = c.CManager.backend.CreateWatch(key)
-
-	go c.processKVStoreWatchEvents(key)
-
-	return c.changeEventChan, nil
-}
-
-//processKVStoreWatchEvents process the kvStoreEventChan and create ConfigChangeEvent
-func (c *ComponentConfig) processKVStoreWatchEvents(key string) {
-	//In a loop process incoming kvstore events and push changes to changeEventChan
-
-	//call defer backend delete watch method
-
-	for watchResp := range c.kvStoreEventChan {
-		configEvent := newChangeEvent(watchResp.EventType, watchResp.Key, watchResp.Value)
-		c.changeEventChan <- configEvent
-	}
-}
-
-// NewEvent creates a new Event object
-func newChangeEvent(eventType int, key, value interface{}) *ConfigChangeEvent {
-	//var key string
-	evnt := new(ConfigChangeEvent)
-	evnt.ChangeType = eventType
-	evnt.Key = fmt.Sprintf("%s", key)
-	evnt.Value = value
-
-	return evnt
-}
-
-func (c *ComponentConfig) Retreive() (map[string]interface{}, error) {
-	//retrieve the data for componentLabel,configType and configKey
-	//e.g. componentLabel "adapter",configType "loglevel" and configKey "config"
-
-	//construct key using makeConfigPath
-	key := c.makeConfigPath()
-
-	//perform get operation on backend using constructed key
-	data, err := c.CManager.backend.Get(key)
-	if err != nil || data == nil {
-		return nil, err
-	}
-
-	res := make(map[string]interface{})
-	if data.Value != nil {
-		res[data.Key] = data.Value
-
-		dat := populateLogLevel(res)
-		return res, nil
-	}
-	return nil, nil
-
-}
-
-func (c *ComponentConfig) RetreiveAsString() (string, error) {
-	//call  Retrieve method
-	c.CManager.backend.Lock()
-	defer c.CManager.backend.Unlock()
```

Comment:
Should not be doing lock and unlock on Backend yourself. Let it be done by backend methods themselves